kiali.go

```diff
@@ -184,14 +184,11 @@ func validateConfig() error {
 		return fmt.Errorf("server static content root directory does not exist: %v", config.Get().Server.StaticContentRootDirectory)
 	}
 
-	validPathRegEx := regexp.MustCompile(`^\/[a-zA-Z0-9\-\._~!\$&\'()\*\+\,;=:@%/]*$`)
+	validPathRegEx := regexp.MustCompile(`^[a-zA-Z0-9\-\._~!\$&\'()\*\+\,;=:@%/]*$`)
 	webRoot := config.Get().Server.WebRoot
 	if !validPathRegEx.MatchString(webRoot) {
 		return fmt.Errorf("web root must begin with a / and contain valid URL path characters: %v", webRoot)
 	}
-	if webRoot != "/" && strings.HasSuffix(webRoot, "/") {
```

Comment:
@israel-hdez  and @jmazzitelli I am removing this check as for reverse proxy '/' is not needed

Reply:
@inyee786 Are you saying that the reverse proxy CAN have a trailing slash `/` at the end of the configured web_root?

This check ensures that `web_root` never ends with a slash `/` character. We have seen this break in the past when users append "/" to the end of web root (that is why this check is there). In other words, we have seen it where users set the web root to things like `/kiali/` and the system breaks (something about the request processor - I forget the reason why).

So please check other use-cases (other than your reverse proxy use-case) to make sure that change will not break things. Specifically, you will want to test that web_root="/kiali/" works. If it does not work but we still want to allow this (for those with reverse proxies that end the web_root with a `/`) we will  need to log a WARN message here to let people know about potential issues.

Note also that the operator will strip any ending slash that is found in web_root - see: https://github.com/kiali/kiali/blob/master/operator/roles/kiali-deploy/tasks/main.yml#L86-L91

So if we are to change this behavior, we will need the operator to log a debug message here and not change web_root if, in fact, we can allow web_root to end with a slash. So that operator ansible script will need to change as well.

Reply:
No, I am saying that it should not contain trailing slash `/` at the end of webroot but
here 
https://github.com/kiali/kiali/blob/b8fc8fe782042fea27db363c3a1d16b9ce4d3757/kiali.go#L271  it getting updated with `/` at the end

Reply:
> No, I am saying that it should not contain trailing slash `/` at the end of webroot but here

Yes, that is what this code does. I am confused then. This code:

```
	if webRoot != "/" && strings.HasSuffix(webRoot, "/") {
```

Is merely confirming that the user did not configure web_root with the trailing slash. Thus eliminating the possibility that we end up with `<base href=/base_root_here//`> (notice the duplicate trailing slash). What is the purpose of removing that check `strings.HasSuffix(webRoot, "/")` ? It sounds like the reverse proxy wants the same as the normal root configurations (where the user should not put a trailing slash because Kiali will add one for you - this eliminates duplicate trailing slashes).